src/main/java/com/epam/ta/reportportal/dao/IntegrationRepository.java

```diff
@@ -38,16 +40,80 @@
 	Optional<Integration> findByIdAndProjectId(Long id, Long projectId);
 
 	/**
-	 * Retrieves given project's integrations
+	 * Retrieve given project's integrations
 	 *
 	 * @param projectId ID of project
 	 * @return Found integrations
 	 */
 	List<Integration> findAllByProjectId(Long projectId);
 
+	/**
+	 * Retrieve all {@link Integration} by project ID and integration type
+	 *
+	 * @param projectId       {@link com.epam.ta.reportportal.entity.project.Project#id}
+	 * @param integrationType {@link IntegrationType}
+	 * @return The {@link List} of the {@link Integration}
+	 */
+	List<Integration> findAllByProjectIdAndType(Long projectId, IntegrationType integrationType);
+
+	/**
+	 * Delete all {@link Integration} by integration type ID
+	 *
+	 * @param typeId {@link IntegrationType#id}
+	 */
+	void deleteAllByTypeId(Long typeId);
+
+	/**
+	 * Retrieve all {@link Integration} with {@link Integration#project} == null by integration type
+	 *
+	 * @param integrationType {@link Integration#type}
+	 * @return @return The {@link List} of the {@link Integration}
+	 */
+	@Query(value = "SELECT i FROM Integration i WHERE i.project IS NULL AND i.type = :integrationType")
+	List<Integration> findAllGlobalByType(@Param("integrationType") IntegrationType integrationType);
+
+	/**
+	 * Find BTS integration by BTS url, BTS project name and Report Portal project id
+	 *
+	 * @param url        Bug Tracking System url
+	 * @param btsProject Bug Tracking System project name
+	 * @param projectId  {@link com.epam.ta.reportportal.entity.project.Project#id}
+	 * @return The {@link Integration} wrapped in the {@link Optional}
+	 */
 	@Query(value = "SELECT i.id, i.enabled, i.project_id, i.creation_date, i.params, i.type, 0 as clazz_ FROM integration i"
 			+ " WHERE (params->'params'->>'url' = :url AND params->'params'->>'project' = :btsProject"
 			+ " AND i.project_id = :projectId) LIMIT 1", nativeQuery = true)
-	Optional<Integration> findByUrlAndBtsProjectAndProjectId(@Param("url") String url, @Param("btsProject") String btsProject,
+	Optional<Integration> findProjectBtsByUrlAndLinkedProject(@Param("url") String url, @Param("btsProject") String btsProject,
```

Comment:
If an integration is a abstraction for all supported integrations, why do we need an explicit method for Bts?
Maybe it is better make bts type as a parameter and add it to the sql query ?